crates/brainfuck_prover/src/components/io/table.rs

```diff
@@ -77,6 +85,51 @@ impl<const N: u32> IOTable<N> {
             self.add_row(dummy_row);
         }
     }
+
+    /// Transforms the [`IOTable`] into a [`TraceEval`], to be committed when
+    /// generating a STARK proof.
+    ///
+    /// The [`IOTable`] is transformed from an array of rows (one element = one step
+    /// of all registers) to an array of columns (one element = all steps of one register).
+    /// It is then evaluated on the circle domain.
+    ///
+    /// # Returns
+    /// A tuple containing the evaluated trace and claim for STARK proof.
+    ///
+    /// # Errors
+    /// Returns [`TraceError::EmptyTrace`] if the table is empty.
+    pub fn trace_evaluation(&self) -> Result<(TraceEval, Claim), TraceError> {
+        let n_rows = self.table.len() as u32;
+
+        // If the table is empty, return an error.
+        if n_rows == 0 {
+            return Err(TraceError::EmptyTrace);
+        }
```

Comment:
There could have programs which don't do any input/output so the I/O table might be empty

Reply:
Yeah, I'm not sure on how to handle this case, maybe early return with a default empty TraceEval and a Claim with log_size to 0, wdyt?

Reply:
Yes seems logical to me like this and we can treat this trace in a particular way afterwards if we need for this specific case. Just modified it